students/guillaume/session07/test_html_render.py

```diff
@@ -0,0 +1,65 @@
+from html_render import Element, Body, Para, HTML
+
+
+def test_new_element():
+    el_object = Element()
+    el_object = Element('content')
+
+
+def test_add_content():
+    el_object = Element()
+    el_object = Element()
+    assert el_object.content == []
+
+
+def test_adding_empty_string():
+    el_object = Element('')
+    assert el_object.content == ['']
+
+
+def test_append_string():
+    el_object = Element('spam, spam. spam')
+    el_object.append(', wonderful spam')
+    assert el_object.content == ['spam, spam. spam', ', wonderful spam']
+
+
+def test_tag_exists():
+    assert Element.tag == 'html'
+    el_object = Element('spam, spam, spam')
+    assert el_object.tag == 'html'
+
+
+def test_indent_exists():
+    assert Element.indent == '  '
+
+
+def test_Redner():
+    my_stuff = 'spam, spam, spam'
+    el_object = Element(my_stuff)
+    more_stuff = 'eggs, eggs, eggs'
+    el_object.append(more_stuff)
+    with open('test1', 'w') as out_file:
+        el_object.render(out_file)
+    with open('test1', 'r') as in_file:
+        contents = in_file.read()
+    assert contents.startswith('<html>')
+    assert contents.endswith('</html>')
+    assert my_stuff in contents
+    assert more_stuff in contents
+
+
+def test_body_tag():
```

Comment:
I'll talk about this in class, but you don't want to test too much implementation detail --and having the tag attribute is an implementation detail.